scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2294,7 +2294,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
         val body: Term = token match {
           case _ if dialect.allowTryWithAnyExpr => expr()
           case LeftParen() => inParensOrUnit(expr())
-          case LeftBrace() | Indentation.Indent() => block()
```

Comment:
Turns out `try {...}.select` was also problematic. `expr()` is enough here, since it will also catch a block. I checked anything invoking `block()` or `blockExpr()` if doesn't have the same issue.

Reply:
`try {...}.select` is only allowed when `allowTryWithAnyExpr` is set (see line 2295). so it looks like you don't need to change this line.

Reply:
Och, it seems we are using Scala211 dialect as default in a lot places :thinking: 

We could try setting the dialect based on the current version of Scala, which would test much more things. I will follow up with another PR.